Training/interface/DataLoader_main.h

```diff
@@ -472,7 +472,7 @@ class DataLoader {
                                     std::make_index_sequence<nFeaturesTypes>{});
 
         auto fillGrid = [&](auto _feature_idx, float value) {
-          if(static_cast<int>(_feature_idx) < 0) return;
+          if(static_cast<int>(_feature_idx) < 0) return; // To skip NANs, in cases where a few are known to exist: if((static_cast<int>(_feature_idx) < 0) || !(std::isfinite(value))) return;
```

Comment:
If there are some `pfCand_gamma` variables with NaN which are not screened, it should be done in the dataLoader code. We can have a single if block for all track-related pf gamma variables. E.g.
```c++
if(tau.pfCand_vertex_z.at(pfCand_idx) != tau.pv_z) {
  fillGrid(Br::pfCand_gamma_vertex_dx, tau.pfCand_vertex_x.at(pfCand_idx) - tau.pv_x);
  fillGrid(Br::pfCand_gamma_vertex_dy, tau.pfCand_vertex_y.at(pfCand_idx) - tau.pv_y);
  fillGrid(Br::pfCand_gamma_vertex_dz, tau.pfCand_vertex_z.at(pfCand_idx) - tau.pv_z);
  fillGrid(Br::pfCand_gamma_vertex_dt, tau.pfCand_vertex_t.at(pfCand_idx) - tau.pv_t);
  fillGrid(Br::pfCand_gamma_vertex_dx_tauFL, tau.pfCand_vertex_x.at(pfCand_idx) - tau.pv_x -
                                             tau.tau_flightLength_x);
  fillGrid(Br::pfCand_gamma_vertex_dy_tauFL, tau.pfCand_vertex_y.at(pfCand_idx) - tau.pv_y -
                                             tau.tau_flightLength_y);
  fillGrid(Br::pfCand_gamma_vertex_dz_tauFL, tau.pfCand_vertex_z.at(pfCand_idx) - tau.pv_z -
                                             tau.tau_flightLength_z);
} else {
  fillGrid(Br::pfCand_gamma_vertex_dx_tauFL, tau.pv_x - tau.tau_flightLength_x);
  fillGrid(Br::pfCand_gamma_vertex_dy_tauFL, tau.pv_y - tau.tau_flightLength_y);
  fillGrid(Br::pfCand_gamma_vertex_dz_tauFL, tau.pv_z - tau.tau_flightLength_z);
}
```

Reply:
I'm wondering if it's really necessary to ignore the `dx`, `dy`, `dt` varables too if only `dz` is bad (probably if you can't trust those values in that case?), but I guess it doesn't matter because it's only for a few taus anyway.
One comment: `a != b` will always be true if `a` and/or `b` are NAN, so I'll check with `isfinite` instead.

Reply:
For quite NaNs, result of all boolean operators is `false`. E.g. `if(!(x == x))` is a standard way to check for NaNs. But ok, `isfinite` is more readable indeed :)